lib/krypto/challenge.rb

```diff
@@ -6,7 +6,7 @@
 
 module Krypto
   class Challenge
-    def self.generate(signing_key, challenge_id, challenge_data, request_data, timestamp: Time.now)
+    def self.generate(signing_key, challenge_id, challenge_data, request_data, timestamp = Time.now)
```

Comment:
Idiomatically, I'm thinking about this as having 4 required arguments, and then 1 optional timestamp. Potentially, in the future, there may be more optional arguments. Knowing that, do you want to move `timestamp` to positional?

Reply:
Knowing that I would expect the method to look something like this

```ruby
def self.generate(signing_key, challenge_id, challenge_data, request_data, **options)
   options = { timestamp: Time.now.utc.to_i }.merge(options)
   ...
end
```

Reply:
I'm not sure we need ruby 1.9.

I think `timestamp: Time.now` gives us a better path to get to `**options`.

Thinking about it, I'm uncomfortable with optional positional arguments. I'd suggest moving to all named style, or leaving as is.